src/api/app/assets/javascripts/webui2/project_monitor.js

```diff
@@ -6,18 +6,75 @@ function setAllLinks(event) {
   $(this).closest('.dropdown-menu').find('input').prop('checked', event.data.checked);
 }
 
-function setupProjectMonitor() { // jshint ignore:line
+function statusCell(meta, statusHash, tableInfo, projectName, packageName) {
+  var info = tableInfo[meta.col - 1];
+  var repository = info[0];
+  var architecture = info[1];
+  var status = statusHash[repository][architecture][packageName] || {};
+  var code = status.code;
+  if (code === undefined) { return null; }
+
+  var output = '<a ';
+  if (['succeeded', 'failed', 'building'].includes(code)) {
+    var url = '/package/live_build_log/' + projectName + '/' + packageName + '/' + repository + '/' + architecture;
+    output += 'href="' + url + '"';
+  }
+  else {
+    var id = meta.row + '-' + meta.col;
+    output += 'href="javascript:void(0);" id="' + id + '"';
+
+    if (status.details !== undefined) {
+      if (code === 'scheduled') klass = 'text-warning';
+      output += ' data-content="' + status.details + '" data-placement="right" data-toggle="popover"';
+    }
+  }
+  var klass = 'build-state-' + code;
+  output += ' class="' + klass + '">' + code + '</a>';
+  return output;
+}
```

Comment:
Have you considered to using JQuery to create the element? I think that would allow us to add add data attributes, classes and the href in a simpler way using JQuery methods (using `data()` and `addClass`).

Reply:
I am not sure what you mean, can you please elaborate?

Reply:
It might have some performance impact to use JQuery here, but if it makes the code considerably more readable and doesn't make it to slow it should be worth it. We could measure it

Reply:
This would have performance consequences, I don't find it more readable and it is out of the scope of this PR. Remember this PR is needed to continue improving the efficiency of the page.

Reply:
I understand know. This would have performance consequences, I don't find it more readable, it may bring problems with escaping urls and it is out of the scope of this PR. Remember this PR is needed to continue improving the efficiency of the page.

Reply:
Instead of concatenating strings (which is what we do with `output` now), this is how you could do it with jQuery:

```javascript
$('<div>', { class: 'lala' });
// returns Object [ div.lala ]
```

[Documentation](https://api.jquery.com/jQuery/#jQuery2)

Reply:
@dmarcoux

> could we try if using jQuery is really a hit in the performance

You can try it... But I still don't get what we gain with this. You can do it with jQuery and you can also do it with a string. What do we gain from changing it? This is already way much faster and it is needed to continue the performance improvement. I don't think this discussion belongs to this PR

Reply:
> We can try it... But I still don't get what we gain with this.

We would gain more readable code :smiley:

I already collected some different ways how we could create the element and plan to create some benchmark for it. What I don't like about this code is that it mixes how the data is collected and how it is presented.

But I am fine doing this in a second step, given that the sprint is almost over.

Reply:
> But I am fine doing this in a second step, given that the sprint is almost over.

Me too